src/jquery.xmleditor.js

```diff
@@ -595,65 +604,79 @@ $.widget( "xml.xmlEditor", {
-			'contentType' : "application/xml",
-			'type' : "POST",
-			'data' : xmlString,
-			success : function(response) {
-				// Process the response from the server using the provided response handler
-				// If the result of the handler evaluates true, then it is assumed to be an error
-				var outcome = self.options.submitResponseHandler(response);
-				
-				if (!outcome) {
-					// 
-					self.xmlState.changesCommittedEvent();
-					self.clearProblemPanel();
-				} else {
-					self.xmlState.syncedChangeEvent();
+	submitXML: function(url) {
+		if (url){
+			if (this.textEditor.active) {
+				try {
+					this.setXMLFromEditor();
+				} catch (e) {
+					this.xmlState.setDocumentHasChanged(true);
 					$("." + submissionStatusClass).html("Failed to submit<br/>See errors at top").css("background-color", "#ffbbbb").animate({backgroundColor: "#ffffff"}, 1000);
-					self.addProblem("Failed to submit xml document", outcome);
-				}
-			},
-			error : function(jqXHR, exception) {
-				if (jqXHR.status === 0) {
-					alert('Not connect.\n Verify Network.');
-				} else if (jqXHR.status == 404) {
-					alert('Requested page not found. [404]');
-				} else if (jqXHR.status == 500) {
-					alert('Internal Server Error [500].');
-				} else if (exception === 'parsererror') {
-					alert('Requested JSON parse failed.');
-				} else if (exception === 'timeout') {
-					alert('Time out error.');
-				} else if (exception === 'abort') {
-					alert('Ajax request aborted.');
-				} else {
-					alert('Uncaught Error.\n' + jqXHR.responseText);
+					this.addProblem("Cannot submit due to invalid xml", e);
+					return false;
 				}
 			}
-		});
+			// convert XML DOM to string
+			var xmlString = this.xml2Str(this.xmlState.xml);
+			$("." + submissionStatusClass).html("Submitting...");
+			var self = this;
+			$.ajax({
+				'url' : url,
+				'contentType' : "application/xml",
+				'type' : "POST",
+				'data' : xmlString,
+				success : function(response) {
+					// Process the response from the server using the provided response handler
+					// If the result of the handler evaluates true, then it is assumed to be an error
+					var outcome = self.options.submitResponseHandler(response);
+	                        if ($("#errors").is(":visible")){
+	                            $("#errors").hide();
+	                         }
+	                         if ($("." + submissionStatusClass).hasClass("alert alert-danger")){
+	                         	$("." + submissionStatusClass).removeClass("alert alert-danger")
+	                         }
+					if (!outcome) {
+						self.xmlState.changesCommittedEvent();
+						self.clearProblemPanel();
+					} else {
+						self.xmlState.syncedChangeEvent();
+						$("." + submissionStatusClass).html("Failed to submit<br/>See errors at top").css("background-color", "#ffbbbb").animate({backgroundColor: "#ffffff"}, 1000);
+						self.addProblem("Failed to submit xml document", outcome);
+					}
+					if (response.localName){
```

Comment:
Sure, 

The success might be returning to a page where errors had been, and so the first part of the code checks for them - for standard bootstrap, rails classes - and removes them if they're found. We're using it in a Rails app.

The code in the else of the !outcome seems unnecessary, but had been there, so I left it. Maybe it's for a situation where the server returns a 200 but not some required data? 

The last part is checking for a url to redirect to, if it's returned. Our code does this, our users are potentially editing many records in one batch, and we return them to the batch upon a successful submission.